core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -32,17 +32,15 @@
 import org.apache.iceberg.types.Types;
 import org.assertj.core.api.Assertions;
 import org.assertj.core.api.recursive.comparison.RecursiveComparisonConfiguration;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
-public class TestManifestReader extends TableTestBase {
-  @Parameterized.Parameters(name = "formatVersion = {0}")
-  public static Object[] parameters() {
-    return new Object[] {1, 2};
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ParameterizedTestExtension.class)
```

Comment:
this should be part of the base class. Also the base + subclass shouldn't have a constructor anymore

Reply:
@lisirrx I've opened https://github.com/apache/iceberg/pull/9424 to fix the parameterization as that should live in the base class. If you're ok with the changes I can push them to your branch

Reply:
I found that some classes, for example `ScanPlanningAndReportingTestBase` and `TestCommitReporting` only use value `2` for formatVersion(in their constructor), I'm not sure if it is ok for such classes.

Reply:
those are still parameterized test, but using only a single version in this case

Reply:
Ok, I got it. I have tried to override the `parameters()` with `@Parameters` in subclass and it works. 
You can push the changes to my branch. Thanks for pointing this out!

Reply:
@nastra I noticed you have merged #9424, so should this pr be closed?
BTW, Should #9085 be split into multiple tasks? I want to work on it but there are too many tests.

Reply:
@lisirrx yes this PR can be closed and we can just open multiple PRs referencing #9085 (without having to create subtasks)

Reply:
ok, I'll close this one.